app/services/products_renderer.rb

```diff
@@ -0,0 +1,92 @@
+require 'open_food_network/scope_product_to_hub'
+
+class ProductsRenderer
+  class NoProducts < RuntimeError; end
+
+  def initialize(distributor, order_cycle, customer, args = {})
+    @distributor = distributor
+    @order_cycle = order_cycle
+    @customer = customer
+    @args = args
+  end
+
+  def products_json
+    raise NoProducts unless order_cycle && distributor && products
```

Comment:
and this what feels broken to me. We say `NoProducts` when what we really mean is some custom flavor of [ArgumentError[(https://apidock.com/ruby/v2_1_10/ArgumentError).

Besides, I would raise it from the `initialize` method right away.